operator/controllers/resources/utils.go

```diff
@@ -0,0 +1,68 @@
+package resources
+
+import (
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/trace"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	ConditionReasonOriginLabelNotMatching = "OriginLabelNotMatching"
+	ConditionReasonOriginLabelMatching    = "OriginLabelMatching"
+	ConditionReasonNewResource            = "NewResource"
+	ConditionTypeTeleportResourceOwned    = "TeleportResourceOwned"
+)
+
+// isResourceOriginKubernetes reads a teleport resource metadata, searches for the origin label and checks its
+// value is kubernetes.
+func isResourceOriginKubernetes(resource types.Resource) bool {
+	metadata := resource.GetMetadata()
+	if label, ok := metadata.Labels[types.OriginLabel]; ok {
+		return label == types.OriginKubernetes
+	}
+	return false
+}
+
+func checkOwnership(existingResource types.Resource, setCondition func(condition metav1.Condition) error) error {
+	if existingResource != nil {
```

Comment:
:nail_care: 
Removing the `setCondition` method should help with readability and reasoning
If we want to go a step further, I think we can define the 3 possible conditions as constants and just return them depending on the `if`s we already have

We might have to change things a bit because of the `AlreadyExists without Kube label` branch
We can return the condition and depending on the reason, return a specific error (in this case `trace.AlreadyExists`)